### PR TITLE
Needs #asString in MCFileTreeGitRepository>>basicStoreVersion:

### PR DESCRIPTION
**Pull request by cdlm**
Either that, or Komitter should not use a Text for the commit message… otherwise it commits to the local package cache but not to the GitFileTree working copy.

### CODE REVIEW

**Comment by cdlm** [comment]
Wait… 200 files changed… what's this mess 😱 

**edit** I had created the MC repo in the wrong directory… however after resetting and trying to redo the commit I still get changes to 25 json files etc. I give up.

**Comment by ThierryGoubier** [comment]
Ok. It seems a few things have been happening inside Pharo 6 which introduces noise among changes:
- JSON output is formatted differently.
- In methodProperties.json, the order of class/instance is reversed.
- in methodProperties.json, selectors are now ordered alphabetically.

i.e. the consequence of replacing the json reader/writer inside FileTree by the Pharo6 one. And git idea of handling json files as text...

@cdlm : don't you think that convertToSystemString could also work on Text instances? Or that Komitter should have a better mastery of the type it manipulates? Because, now, with that asString, any tool which makes a mistake on the object type for the version message will not be seen, with a silent conversion to a String.

File: repository/MonticelloFileTree-Git.package/MCFileTreeGitRepository.class/instance/basicStoreVersion..st [hunk 1]
```diff
@@ -24,7 +24,7 @@ basicStoreVersion: aVersion
 					[ :n | stream nextPut: '--author=' , (n at: #name) , ' ' , (n at: #email) ].
 			stream
 				nextPut: '-m';
-				nextPut: (self escapeForShell: aVersion info message convertToSystemString);
+				nextPut: (self escapeForShell: aVersion info message asString convertToSystemString);
```
Comment (ThierryGoubier) [inline repository/MonticelloFileTree-Git.package/MCFileTreeGitRepository.class/instance/basicStoreVersion..st:27]: #asString will convert anything to a string representation. For a Text instance, this is what we want. But for other objects, is this really what we expect, or is it a case of silencing type errors?

**Comment by cdlm** [comment]
I agree. I find it strange that Komitter sends a `Text` as a commit message, maybe that would make sense if there was a way to convert that back to markdown or something for commits in git, but I think I'd prefer if Komitter was sending a `String` in the first place.

`convertToSystemString` seems a bit low-level: converting a `Text` could involve some formatting (think of ANSI escapes, or how to convert paragraph breaks)

Maybe @demarey will know better ?

**Comment by ThierryGoubier** [comment]
Ok, taken care of via 908afda (added a type check to ensure the version info message is a string)

**Pull request closed without merging**